pkg/upgrade/upgrade.go

```diff
@@ -151,27 +153,39 @@ func DriverToolkitVersion(entries []string, info map[string]NodeVersion) (map[st
 	for _, entry := range entries {
 
 		log.Info("History", "entry", entry)
-		var layer v1.Layer
-		if layer = registry.LastLayer(entry); layer == nil {
+
+		var (
+			err   error
+			layer v1.Layer
+		)
+
+		layer, err = registry.LastLayer(entry)
+		if err != nil {
+			return nil, err
+		}
+
+		if layer == nil {
 			continue
 		}
 		// For each entry we're fetching the cluster version and dtk URL
-		version, imageURL := registry.ReleaseManifests(layer)
+		version, imageURL, err := registry.ReleaseManifests(layer)
 		if version == "" {
-			exit.OnError(errors.New("Could not extract version from payload"))
+			return nil, fmt.Errorf("could not extract version from payload: %w", err)
 		}
 
 		if imageURL == "" {
 			warn.OnError(errors.New("No DTK image found, DTK cannot be used in a Build"))
 			return info, nil
 		}
 
-		if layer = registry.LastLayer(imageURL); layer == nil {
-			exit.OnError(errors.New("Cannot extract last layer for DTK from: " + imageURL))
+		if layer, err = registry.LastLayer(imageURL); layer == nil {
+			return nil, fmt.Errorf("cannot extract last layer for DTK from %s [%w]", imageURL, err)
```

Comment:
Maybe use the same format as the rest of wrapping? `some text wrapper: original error`

Reply:
Right, back to the regular format.